rosbag2_transport/src/rosbag2_transport/recorder.cpp

```diff
@@ -106,13 +106,22 @@ void Recorder::subscribe_topics(
 
 void Recorder::subscribe_topic(const rosbag2::TopicMetadata & topic)
 {
+  if(writer_) {
+    writer_->create_topic(topic);
```

Comment:
wouldn't it be sufficient to call `writer_->create_topic(topic)` before we call `subscriptions_.push_back(subscriptions)` ?

Reply:
I think so. But, don't you feel we may need a fallback when subscription creation fails and the db still has an entry for the same. At the same time, when the subscription is started again, the race would still exist. So may be we may have to clean up the db, so added the remove api.

Reply:
I agree with your last point.